options.go

```diff
@@ -62,6 +62,9 @@ type CloneOptions struct {
 	// within, using their default settings. This option is ignored if the
 	// cloned repository does not have a worktree.
 	RecurseSubmodules SubmoduleRescursivity
+	// ShallowSubmodules limit cloning submodules to the 1 level of depth.
+	// It matches the git command --shallow-submodules.
+	ShallowSubmodules bool
```

Comment:
Please change this to `Depth` instead, to align with the API exposed in other options (e.g. `SubmoduleUpdateOptions.Depth`). Apologies for not picking this up on the first review.

Reply:
Just to understand the philosophy behind the project better - do we strive to provide the 1 to 1 mapping between features in `git` vs `go-git`? Or do we try to provide the same functionality, but we also make the functionality more general if possible?

Why I'm asking this question is because the code above is a direct mapping of the `--shallow-submodules` from git which only allows for the depth of 1 [[git source code](https://github.com/git/git/blob/fe86abd7511a9a6862d5706c6fa1d9b57a63ba09/builtin/clone.c#L760-L761)]. To the best of my understanding there's no option in git to define the depth level of submodules.

Reply:
My understanding is that `go-git` aims to reflect the behaviour and naming convention of upstream `git`, but sometimes this is not a 1-to-1 mapping given the different characteristics of a CLI vs a library.

I think this is quite a specific case. As `git submodule` do have a `--depth`, but when it comes down to `git clone` it has `--shallow-submodules` instead. But realistically they do slightly different things, so based on that, let's go ahead with your changes as is.